### PR TITLE
feat(web): Adding resolve pipeline template endpoint

### PR DESCRIPTION
**Pull request by robzienert**
Endpoint resolves & merges a template inheritance tree by source. This will be used to help Deck render variable bindings while configuring a pipeline template.

@spinnaker/reviewers PTAL

### CODE REVIEW

File: orca-web/src/main/groovy/com/netflix/spinnaker/orca/controllers/PipelineTemplateController.groovy [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+    return [message: e.message, status: HttpStatus.BAD_REQUEST]
+  }
+
+  @ResponseStatus(HttpStatus.BAD_REQUEST)
+  @ExceptionHandler(TemplateLoaderException)
+  Map templateLoadingErrorHandler(TemplateLoaderException e) {
+    return [message: e.message, status: HttpStatus.BAD_REQUEST]
+  }
+
+  static class InvalidTemplateSourceException extends RuntimeException {
```
Comment (danielpeach) [inline orca-web/src/main/groovy/com/netflix/spinnaker/orca/controllers/PipelineTemplateController.groovy]: Could use `@InheritConstructors` here.

**Comment by jeyrschabu** [comment]
LGTM other than the comment above

File: orca-web/src/main/groovy/com/netflix/spinnaker/orca/controllers/PipelineTemplateController.groovy (hunk 1 above)
Comment (robzienert) [inline orca-web/src/main/groovy/com/netflix/spinnaker/orca/controllers/PipelineTemplateController.groovy]: Oh Groovy, you so magical. Thanks.

File: orca-web/src/main/groovy/com/netflix/spinnaker/orca/controllers/PipelineTemplateController.groovy [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+@ConditionalOnProperty("pipelineTemplate.enabled")
+@RestController
+@Slf4j
+class PipelineTemplateController {
+
+  @Autowired
+  PipelineTemplateService pipelineTemplateService
+
+  @RequestMapping(value = "/pipelineTemplate", method = RequestMethod.GET)
+  PipelineTemplate getPipelineTemplate(@QueryParam("source") String source) {
```
Comment (danielpeach) [inline orca-web/src/main/groovy/com/netflix/spinnaker/orca/controllers/PipelineTemplateController.groovy]: I see `RequestParam` in a lot of places - any reason to use `QueryParam` instead?

Comment (robzienert) [inline orca-web/src/main/groovy/com/netflix/spinnaker/orca/controllers/PipelineTemplateController.groovy]: Updated. Turns out `QueryParam` is JAX-RS. It works, but probably preferential to use the Spring `RequestParam` instead.

**Pull request merged**